### ISSUE TITLE
Provide a field with the actual class name of a default exported class 

### ISSUE THREAD

**Issue by Matsuuu** [webcomponents/custom-elements-manifest#72]
As per the spec, (https://github.com/webcomponents/custom-elements-manifest/blob/master/schema.d.ts#L85-L96), a default-exported class gets named "default". However in the Schema's Reference -type,  used by the "superclass" property for example (https://github.com/webcomponents/custom-elements-manifest/blob/master/schema.d.ts#L149-L153), a "name" field is provided, and is always populated by the actual class name of the Javascript class.

When creating tooling, it would be useful to be able to look at the declarations of the superclass, but this is impossible with the current schema as if the class is default exported, the name is not found anywhere on the class' schema.

### Proposed solution

Add a "className" or similiar field to the JavaScriptExport type. This would always be populated with the actual classname of the class, instead of "default" in default exported classes

**Comment by Matsuuu** [comment]
Example case:

https://custom-elements-manifest.netlify.app/?source=CmV4cG9ydCBkZWZhdWx0IGNsYXNzIE15RWxlbWVudCBleHRlbmRzIEhUTUxFbGVtZW50IHt9CgpleHBvcnQgY2xhc3MgTXlFeHRlbmRpbmdFbGVtZW50IGV4dGVuZHMgTXlFbGVtZW50IHt9CmN1c3RvbUVsZW1lbnRzLmRlZmluZSgibXktZXh0ZW5kaW5nLWVsZW1lbnQiLCBNeUV4dGVuZGluZ0VsZW1lbnQpOw%3D%3D&library=vanilla

If we have a MyElement class we want to extend our element from, and then we want to backtrack from MyExtendingElement's "superclass.name" to find the declarations of it, we can't find it with the current information, as the name "MyElement" is not present in the declaration at all.

```json
"declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "default",
          "superclass": {
            "name": "HTMLElement"
          },
          "customElement": true
        },
        {
          "kind": "class",
          "description": "",
          "name": "MyExtendingElement",
          "superclass": {
            "name": "MyElement",
            "module": "src/my-element.js"
          },
          "tagName": "my-extending-element",
          "customElement": true
        }
      ],
```

**Comment by thepassle** [comment]
I've ran into the situation myself a couple of times as well, where I need to find a class that is exported as default. I do think this is currently a limitation of the schema, and we should improve on this

**Comment by justinfagnani** [comment]
The way this should work is the same for any export that renames the item: the export has a name that's independent of the declaration name.

```ts
export default class MyElement {}
```

should result in an export named "default" and a reference to a class declaration named "MyElement".

Same with a rename:
```ts
class MyElementImpl {}
export {MyElementImpl as MyElement}
```

should result in an export named "MyElement" and a reference to a class declaration named "MyElementImpl".

From the playground it looks like the analyzer is getting this wrong.

This:
```json
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "default",
          "superclass": {
            "name": "HTMLElement"
          },
          "customElement": true
        },
```

Should be:
```json
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "MyElement",
          "superclass": {
            "name": "HTMLElement"
          },
          "customElement": true
        },
```

iow, a declaration should never have the name "default", unless someone actually named a class/function/variable that.

**Comment by justinfagnani** [comment]
Looks like this is being fixed in the Open WC analyzer. Closing here.

**Comment by thepassle** [comment]
This was corrected in @custom-elements-manifest/analyzer v0.4.7 👍
